gatsby-node.js

```diff
@@ -34,7 +34,7 @@ exports.createPages = ({ graphql, boundActionCreators }) => {
           path: edge.node.slug, // required
           component: blogPost,
           context: {
-            slug: edge.node.slug,
+            slug: { eq: edge.node.slug },
```

Comment:
This is... bad. There's a lot of functionality in these nested components. Is this a constraint of the compiler or the client runtime?